node/src/components/consensus/highway_core/state.rs

```diff
@@ -503,6 +498,35 @@ impl<C: Context> State<C> {
         }
         equivocators
     }
+
+    /// Returns an error if the vote's round exponent was changed away from the cited votes' median
+    /// by more than the allowed spread.
+    fn check_round_exp_spread(&self, wvote: &WireVote<C>) -> Result<(), VoteError> {
+        let panorama = &wvote.panorama;
+        if let Observation::Correct(vh) = &panorama[wvote.creator] {
+            if wvote.round_exp >= self.vote(vh).round_exp {
+                return Ok(()); // Round exponent was not decreased.
+            }
+        }
+        let upper_re = match wvote.round_exp.checked_add(self.params.round_exp_spread()) {
+            Some(upper_re) => upper_re,
+            None => return Ok(()), // The median can't be greater than u8::MAX.
+        };
+        // Check that round exponent plus allowed is >= median.
```

Comment:
I don't understand what _plus allowed_ means here – we're not adding anything.

Reply:
Me neither. :sweat_smile: I think I moved words around until it didn't make sense anymore. I'll fix it.